.github/workflows/publish-on-tag.yml

```diff
@@ -131,8 +131,10 @@ jobs:
       contents: read
       packages: write
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
```

Comment:
will it work without checkout?

Reply:
I've read somewhere the checkout step wasn't necessary because `docker/setup-buildx-action@v3` does that, but I'm not finding this information anymore and the example there uses checkout.
Going to restore it.